internal/e2e: GAE E2E tests failing

```
--- FAIL: TestGo11xStatic (0.08s)
    standard_test.go:62: could not deploy static: could not read service: open github.com/GoogleCloudPlatform/golang-samples/appengine/go11x/static/app.yaml: no such file or directory
--- FAIL: TestHelloWorld (0.08s)
    standard_test.go:62: could not deploy hw: could not read service: open github.com/GoogleCloudPlatform/golang-samples/appengine/go11x/helloworld/app.yaml: no such file or directory
FAIL
FAIL	github.com/GoogleCloudPlatform/golang-samples/internal/e2e	0.168s
```

--- a/internal/testutil/testutil.go
+++ b/internal/testutil/testutil.go
@@ -21,6 +21,7 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
 
 	"golang.org/x/tools/go/packages"
@@ -89,11 +90,22 @@ func testContext() (Context, error) {
 		return tc, noProjectID
 	}
 
-	pkgs, err := packages.Load(&packages.Config{Mode: packages.NeedName}, "github.com/GoogleCloudPlatform/golang-samples")
-	if err != nil {
-		return tc, fmt.Errorf("Could not find golang-samples: %v", err)
+	cfg := &packages.Config{
+		Mode:  packages.NeedName | packages.NeedFiles,
+		Tests: true,
 	}
-	tc.Dir = pkgs[0].PkgPath
+	pkgs, err := packages.Load(cfg, "github.com/GoogleCloudPlatform/golang-samples")
+	if err != nil {
+		return tc, fmt.Errorf("could not find golang-samples: %v", err)
+	}
+	for _, pkg := range pkgs {
+		if len(pkg.GoFiles) > 0 && strings.HasSuffix(pkg.GoFiles[0], ".go") {
+			tc.Dir = filepath.Dir(pkg.GoFiles[0])
+		}
+	}
+	if tc.Dir == "" {
+		return tc, fmt.Errorf("could not find golang-samples directory")
+	}
 
 	return tc, nil
 }

--- a/internal/testutil/testutil.go
+++ b/internal/testutil/testutil.go
@@ -98,8 +98,15 @@ func testContext() (Context, error) {
 	if err != nil {
 		return tc, fmt.Errorf("could not find golang-samples: %v", err)
 	}
+	// packages.Load returns multiple values, some with files and some without.
+	// Some of the files are generated as part of the build and some are the
+	// normal Go source files we're looking for.
+	// We can probably assume the one we want is pkgs[2], but loop through
+	// looking for the one we want in case it ever changes.
 	for _, pkg := range pkgs {
 		if len(pkg.GoFiles) > 0 && strings.HasSuffix(pkg.GoFiles[0], ".go") {
+			// Use the directory of a file in the root package as the module
+			// root directory.
 			tc.Dir = filepath.Dir(pkg.GoFiles[0])
 		}
 	}

